lightning/src/ln/channel.rs

```diff
@@ -3675,6 +3675,24 @@ impl<Signer: Sign> Channel<Signer> {
 		Ok((None, timed_out_htlcs))
 	}
 
+	pub fn funding_transaction_unconfirmed(&mut self) -> Result<(), msgs::ErrorMessage> {
+		if self.funding_tx_confirmation_height != 0 {
+			let effective_new_height = self.funding_tx_confirmation_height - 1;
```

Comment:
IIUC, setting "effective_new_height" to "funding_tx_confirmation_height" - 1 will force this latter value to be reset to 0 in `check_get_funding_locked` ? I don't think "effective_new_height" is a compelling name if so maybe "unconf_height" and maybe explain with a code comment that it will latch "funding_tx_confirmation_height" to 0, the effect is 2 function calls under.